lib/Messenger/Handler/AssetUpdateTasksHandler.php

```diff
@@ -68,18 +67,21 @@ private function processDocument(Asset\Document $asset): void
             $asset->save(['versionNote' => 'PDF scan result']);
         }
 
+        $savedNeeded = false;
         $pageCount = $asset->getCustomSetting('document_page_count');
         if (!$pageCount || $pageCount === 'failed') {
-            if ($asset->processPageCount()) {
-                $this->saveAsset($asset);
-            }
+            $savedNeeded |= $asset->processPageCount();
```

Comment:
```suggestion
            $savedNeeded = $asset->processPageCount();
```
@cancan101 Is there a special reason why you use here a Bitwise OR assigment? Because for me that doesn't make sense with 2 booleans